app/services/booking_responder.rb

```diff
@@ -0,0 +1,26 @@
+class BookingResponder
+  def initialize(booking_response)
+    @booking_response = booking_response
+  end
+
+  def respond!
+    if booking_response.slot_selected?
+      accept!
+    else
+      reject!
+    end
+  end
+
+private
+
+  attr_reader :booking_response
+  delegate :visit, to: :booking_response
```

Comment:
To be honest, I don't think it really matters all that much – I see `private` in Ruby as mainly advisory – but I've added `private :visit` below to be sure.